airflow/providers/google/cloud/links/dataproc.py

```diff
@@ -20,99 +20,151 @@
 )
+
 DATAPROC_WORKFLOW_TEMPLATE_LINK = (
-    DATAPROC_BASE_LINK + "/workflows/templates/{region}/{resource}?project={project_id}"
+    DATAPROC_BASE_LINK + "/workflows/templates/{region}/{workflow_template_id}?project={project_id}"
+)
+DATAPROC_WORKFLOW_LINK = (
+    DATAPROC_BASE_LINK + "/workflows/instances/{region}/{workflow_id}?project={project_id}"
 )
-DATAPROC_WORKFLOW_LINK = DATAPROC_BASE_LINK + "/workflows/instances/{region}/{resource}?project={project_id}"
-DATAPROC_BATCH_LINK = DATAPROC_BASE_LINK + "/batches/{region}/{resource}/monitoring?project={project_id}"
+
+DATAPROC_BATCH_LINK = DATAPROC_BASE_LINK + "/batches/{region}/{batch_id}/monitoring?project={project_id}"
 DATAPROC_BATCHES_LINK = DATAPROC_BASE_LINK + "/batches?project={project_id}"
 
 
-class DataprocLink(BaseOperatorLink):
-    """Helper class for constructing Dataproc resource link."""
+class DataprocClusterLink(BaseGoogleLink):
+    """Helper class for constructing Dataproc Cluster Link."""
 
-    name = "Dataproc resource"
-    key = "conf"
+    name = "Dataproc Cluster"
+    key = "dataproc_cluster"
+    format_str = DATAPROC_CLUSTER_LINK
 
     @staticmethod
     def persist(
         context: Context,
-        task_instance,
-        url: str,
-        resource: str,
+        operator_instance: BaseOperator,
+        cluster_id: str,
+        region: str,
+        project_id: str,
     ):
-        task_instance.xcom_push(
-            context=context,
-            key=DataprocLink.key,
-            value={
-                "region": task_instance.region,
-                "project_id": task_instance.project_id,
-                "url": url,
-                "resource": resource,
-            },
+        operator_instance.xcom_push(
+            context,
+            key=DataprocClusterLink.key,
+            value={"cluster_id": cluster_id, "region": region, "project_id": project_id},
         )
 
-    def get_link(
-        self,
-        operator: BaseOperator,
-        *,
-        ti_key: TaskInstanceKey,
-    ) -> str:
-        conf = XCom.get_value(key=self.key, ti_key=ti_key)
-        return (
-            conf["url"].format(
-                region=conf["region"], project_id=conf["project_id"], resource=conf["resource"]
-            )
-            if conf
-            else ""
+
+class DataprocJobLink(BaseGoogleLink):
+    """Helper class for constructing Dataproc Job Link."""
+
+    name = "Dataproc Job"
+    key = "dataproc_job"
+    format_str = DATAPROC_JOB_LINK
+
+    @staticmethod
+    def persist(
+        context: Context,
+        operator_instance: BaseOperator,
+        job_id: str,
+        region: str,
+        project_id: str,
+    ):
+        operator_instance.xcom_push(
+            context,
+            key=DataprocJobLink.key,
+            value={"job_id": job_id, "region": region, "project_id": project_id},
         )
 
 
-class DataprocListLink(BaseOperatorLink):
```

Comment:
The same here, may be add a deprecation warning for the deleted class?

Reply:
changed